rados/conn.go

```diff
@@ -280,6 +281,22 @@ func (c *Conn) DeletePool(name string) error {
 	}
 }
 
+// GetPoolByName returns the ID of the pool with a given name.
+func (c *Conn) GetPoolByName(name string) (int64, error) {
+	if err := c.ensure_connected(); err != nil {
+		fmt.Println("NOT CONNECTED WHOOPS")
+		return 0, err
+	}
+	c_name := C.CString(name)
+	defer C.free(unsafe.Pointer(c_name))
+	ret := int64(C.rados_pool_lookup(c.cluster, c_name))
+	if ret == -C.ENOENT {
```

Comment:
Is ENOENT the only error condition this api call can return? It seems to me like it would be preferable to handle all error codes as errors at the Go level. Compare the code here to https://github.com/ceph/ceph/blob/master/src/pybind/rados/rados.pyx#L1044 (from the python bindings in ceph/ceph)  it explicitly handles ENOENT but also treates all negative values as the errno. I think we should mimic that behavior.

Reply:
That is reasonable. I'll replace the check with `ret < 0` and will not handle `ENOENT` differently.